R/data_rename.R

```diff
@@ -122,14 +157,19 @@ data_rename <- function(data,
     }
   }
 
+  # check if we have "glue" styled replacement-string
+  glue_style <- length(replacement) == 1 &&
+    grepl("{", replacement, fixed = TRUE) &&
+    length(pattern) > 1
```

Comment:
I'm wondering if the `length(pattern) > 1` constraint is useful here. I think it may lead to subtle bugs if someone uses this function like this:
``` r
library(datawizard)
my_fun <- function(cols_to_rename) {
  data_rename(head(mtcars)[, 1:6], cols_to_rename, "new_{col}")
} 

# Token is used
names(my_fun(c("mpg", "drat")))
#> [1] "new_mpg"  "cyl"      "disp"     "hp"       "new_drat" "wt"

# Token is not used
names(my_fun("mpg"))
#> [1] "new_{col}" "cyl"       "disp"      "hp"        "drat"      "wt"
```